### PR TITLE
Debug CPU and Seed

### PR DESCRIPTION
**Pull request by rasaHosseinzadeh**
- Fix exceptions thrown for CPU.
- Fix repetition of seed in loops
- Set seed for knn index.
- Update regression example with lower numbers to make it faster.

### CODE REVIEW

File: src/tabdpt/estimator.py [hunk 1]
```diff
@@ -30,9 +30,9 @@ def __init__(
         compile: bool = True,
     ):
         self.mode = mode
-        self.inf_batch_size = inf_batch_size
         self.device = device or ("cuda" if torch.cuda.is_available() else "cpu")
-        self.use_flash = use_flash
+        self.inf_batch_size = inf_batch_size if self.device == "cuda" else min(inf_batch_size, 16)
```
Comment (anthonycaterini) [inline src/tabdpt/estimator.py:34]: `16` is hardcoded here, should be defined as a constant higher up

File: src/tabdpt/estimator.py [hunk 2]
```diff
@@ -30,9 +30,9 @@ def __init__(
...
+        self.use_flash = use_flash and self.device == "cuda"
```
Comment (anthonycaterini) [inline src/tabdpt/estimator.py:35]: should we give a warning if we aren't using flash attention when the user asks for it?

File: src/tabdpt/estimator.py [hunk 3]
```diff
@@ -46,12 +46,12 @@ def __init__(
...
 
         cfg.env.device = self.device
-        self.model = TabDPTModel.load(model_state=model_state, config=cfg, use_flash=use_flash)
+        self.model = TabDPTModel.load(model_state=model_state, config=cfg, use_flash=self.use_flash)
         self.model.eval()
 
         self.max_features = self.model.num_features
         self.max_num_classes = self.model.n_out
-        self.compile = compile
+        self.compile = compile and self.device == "cuda"
```
Comment (anthonycaterini) [inline src/tabdpt/estimator.py:54]: same question as above.
are we neglecting to compile whenever we are on CPU because it's finicky?

File: tests/reg_example.py [hunk 4]
```diff
@@ -5,9 +5,10 @@
 from tabdpt import TabDPTRegressor
 
 X, y = fetch_california_housing(return_X_y=True)
+X, y = X[:2048], y[:2048]
```
Comment (anthonycaterini) [inline tests/reg_example.py:8]: I don't think this reduction of the dataset should be the default?

**Review by anthonycaterini — CHANGES_REQUESTED** [review]
Had some very basic changes.
Will concurrently test the functionality and report back

**Comment by anthonycaterini** [comment]
functionality works

File: tests/cls_example.py [hunk 5]
```diff
@@ -9,5 +9,5 @@
 
 model = TabDPTClassifier()
 model.fit(X_train, y_train)
-y_pred = model.predict(X_test, n_ensembles=8, temperature=0.8, context_size=1024)
+y_pred = model.predict(X_test, n_ensembles=8, temperature=0.8, context_size=1024, seed=42)
```
Comment (NaGho) [inline tests/cls_example.py:12]: maybe we can lower n_ensembles to 2 here similar to reg for faster run?
